compiler/test/suites/provides.re

```diff
@@ -181,4 +181,19 @@ describe("provides", ({test, testSkip}) => {
     "module Test; provide let rec foo: () => Number = () => 5",
     ("foo", Binaryen.Export.external_function),
   );
+  assertHasWasmExport(
+    "issue_1884_type_provided_later1",
+    "module Test; enum T { A }; let a = A; provide { type T }; provide { a }",
+    ("GRAIN$EXPORT$a", Binaryen.Export.external_global),
+  );
+  assertHasWasmExport(
+    "issue_1884_type_provided_later2",
+    "module Test; enum T { A }; let a = A; provide { a, type T }",
+    ("GRAIN$EXPORT$a", Binaryen.Export.external_global),
+  );
+  assertHasWasmExport(
+    "issue_1884_type_provided_later3",
+    "module Test; enum T { A }; let a = A; provide { a }; provide { type T }",
+    ("GRAIN$EXPORT$a", Binaryen.Export.external_global),
+  );
```

Comment:
This is an unusual-looking case at first but still makes sense to allow imo

Reply:
Im not a major fan of allowing this. but I dont have a major problem with it either.

Reply:
This should 100% be allowed. Thanks for adding a test for it 🤗